### DISCUSSION TITLE
Memory increase on simple counter

### DISCUSSION THREAD

**Discussion by thfsilvab** [slint-ui/slint#5709] [Q&A]
Hello! I'm new and currently learning Slint, checking how well it's going to fit in my use case.

I was going through the docs and building a simple counter just to get started, I know Slint is extremely well optimized and was checking out resource consumption while running the app in release mode, and came across some strange memory spikes when using binding to increase the count.

Starting the memory consumption with 15MB, when increasing the count, the memory keeps going up until 22MB, of course it's not much for a desktop these days, but I was wondering why was that an integer increment would resolve into a +7MB to the running app. Of course it's some lack in my knowledge, and I wanted to fill this gap if someone can help me :)

I worry because the project I'm writing will run in very low-end as well as old computers, and I'm being mindful of the resources being consumed at runtime.

This is the Slint code that's causing this behaviour as well as the environment I'm running the app:
```slint
// Source code is Rust, only has the basics to run the app as shown in the docs

import { Button } from "std-widgets.slint";

export component MainWindow inherits Window {
    title: "Hello World from Slint!!";
    preferred-width: 900px;
    preferred-height: 700px;

    VerticalLayout {
        spacing: 10px;

        Text {
            text: "hello world";
            color: green;
            font-size: 50px;
        }

        HorizontalLayout {
            property <int> count: 0;

            spacing: 10px;
            padding: 15px;

            Text {
                text: count;
            }

            Button {
                text: "+1";
                clicked => {
                    count += 1;
                }
            }

            VerticalLayout {
                spacing: 10px;

                Button {
                    text: "+10";
                    clicked => {
                        count += 10
                    }
                }

                Button {
                    text: "+100";
                    clicked => {
                        count += 100
                    }
                }

                Button {
                    text: "+1000";
                    clicked => {
                        count += 1000
                    }
                }
            }
        }
    }
}
```

![image](https://github.com/user-attachments/assets/0fc2acfd-5d4b-49b7-9f17-7fbe48b5d270)

**Comment by tronical** [comment]
The issue you're seeing is most likely #2768 . It's not a leak per-see, but still annoying. You could switch to Skia as renderer (enable `renderer-skia` in `features`), or if you're really targeting a low-end system, switch to the use of the Slint software renderer.

**Reply by thfsilvab** [reply]
I just tested with multiple combinations of backends and renderes, and unfortunately it still happens with this same code base. I'm currently trying to bind the integer from the Rust code to the UI and see if there's are any differences.
